libs/csm/extensions/csm_ext.py

```diff
@@ -107,9 +112,66 @@ def destroy_prep_deploy_cluster(self, expect_fail=None):
             self.log.info("[Start] Sleep %s", self.update_seconds)
             time.sleep(self.update_seconds)
             self.log.info("[End] Sleep %s", self.update_seconds)
-         
+
             self.log.info("Cleanup: Check cluster status")
             resp = self.ha_obj.poll_cluster_status(self.master)
             assert resp[0], resp[1]
             self.log.info("Cleanup: Cluster status checked successfully")
         self.log.info("[End] Redeploy the setup...")
+
+    def delete_data_pod(self):
+        """
+        Degrade cluster safely by deleting the last data pod from a set
```

Comment:
can we add return docstring?
Also you would always need to delete data pod? IF not if we want to make it generic, maybe u can take pod_name_prefix as input and set default to data pod-- just a suggestion

Reply:
maybe in future if CSM tests need to delete other pods, will make it generic.
or will add another one as the generic, and this function will call that.

Reply:
added docstrings